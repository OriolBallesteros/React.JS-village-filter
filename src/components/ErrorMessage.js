import React, {Fragment} from 'react';

const ErrorMessage = props => {
    return (
        <Fragment>
            <p>Oops! Something went wrong. Try reloading again or show the next message to some developer:</p>
            <p>{props.error.data}</p>
            <p>Url fetched:</p>
            <p>{props.error.config.url}</p>
        </Fragment>
    );
};

export default ErrorMessage;


/**
 * PER RENDERITZAR NOMÃ‰S UN COMPONENT: MIRAR EXPENSLIST.JS DE ANDREW
 * 
 * 
 * <div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses provided</p>
            ) : (
                    props.expenses.map((expense) => (
                        <ExpenseListItem {...expense} key={expense.id} />
                    ))
                )
        }
    </div>
 * 
 * 
 */