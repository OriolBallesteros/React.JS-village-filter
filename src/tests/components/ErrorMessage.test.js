import React from 'react';
import {shallow} from 'enzyme';
import ErrorMessage from '../../components/ErrorMessage';

it('should render ErrorMessage component correctly', ()=>{
    const error = {
        data: 'Something went wrong',
        config: {
            url: 'www.foo.com'
        }
    };
    const wrapper = shallow(<ErrorMessage error={error}/>);

    expect(wrapper).toMatchSnapshot();
});