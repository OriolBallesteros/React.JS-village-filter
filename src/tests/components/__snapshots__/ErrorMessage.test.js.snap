// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ErrorMessage component correctly 1`] = `
<Fragment>
  <p>
    Oops! Something went wrong. Try reloading again or show the next message to some developer:
  </p>
  <p>
    Something went wrong
  </p>
  <p>
    Url fetched:
  </p>
  <p>
    www.foo.com
  </p>
</Fragment>
`;
